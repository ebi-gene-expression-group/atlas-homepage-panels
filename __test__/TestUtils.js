const speciesResources = {
  foo1: `link1`,
  foo2: `link2`,
  foo3: `link3`,
  test1: `endpoint1`,
  test2: `endpoint2`
}

const experimentResources = { 
  Latest: `endpoint3`,
  panel: `test1`
}

const host = `http://`

const SpeciesCardType = () => `I am a species component`

const LatestExperimentCardType = () => `I am a experiment component`

// Stolen from https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/random
const getRandomInt = (min, max) => {
  min = Math.ceil(min)
  max = Math.floor(max)
  return Math.floor(Math.random() * (max - min)) + min //The maximum is exclusive and the minimum is inclusive
}

const latestExperimentEntries = () => {
  return [
    {
      numberOfAssays: 15,
      experimentAccession: `accession1`,
      lastUpdate: `date1`,
      experimentDescription: `description1`,
      species: `species1`
    },
    {
      numberOfAssays: 115,
      experimentAccession: `accession2`,
      lastUpdate: `date2`,
      experimentDescription: `description2`,
      species: `species2`
    },
    {
      numberOfAssays: 1115,
      experimentAccession: `accession3`,
      lastUpdate: `date3`,
      experimentDescription: `description3`,
      species: `species3`
    }
  ]
}

export {getRandomInt, latestExperimentEntries, speciesResources, experimentResources, host, SpeciesCardType, LatestExperimentCardType}