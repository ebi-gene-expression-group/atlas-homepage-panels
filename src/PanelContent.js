import React from 'react'
import PropTypes from 'prop-types'
import HCAlogo from './assets/img/HCA-logo.jpg'

const PanelContent = ({panelName, host, resource, CardType}) => {
  console.log(panelName, CardType)
  switch(panelName) {
    case `Featured`:
      return(    
        <div className="small-4 columns combo text-center padding-top-medium">
          <a href={`${host}${resource[panelName]}`}>
              <img className="margin-bottom-large" src={HCAlogo}  height="400" />
          </a>
        </div>
      )
      break
    default:
      return <CardType host={host} resource={resource[panelName]}/>
  } 
}

PanelContent.propTypes = {
  panelName: PropTypes.string.isRequired,
  host: PropTypes.string.isRequired,
  resource: PropTypes.object.isRequired
}

export default PanelContent
