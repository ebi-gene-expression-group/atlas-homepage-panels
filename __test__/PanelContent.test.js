import React from 'react'
import renderer from 'react-test-renderer'
import Enzyme from 'enzyme'
import { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import PanelContent from '../src/PanelContent'
import {getRandomInt, speciesResources, experimentResources, host, SpeciesCardType, LatestExperimentCardType} from './TestUtils'

Enzyme.configure({ adapter: new Adapter() })

describe(`PanelContent`, () => {
  const randomSpeciesTabName = Object.keys(speciesResources)[getRandomInt(0, Object.keys(speciesResources).length)]
  const randomExperimentTabName = Object.keys(experimentResources)[getRandomInt(0, Object.keys(experimentResources).length)]

  test(`uses a speciesCardContainer for all popular species`, () => {
    const wrapper = shallow(<PanelContent host={host} CardType={SpeciesCardType} resource={speciesResources} panelName={randomSpeciesTabName}/>)
    expect(wrapper.find(LatestExperimentCardType).length).toEqual(0)
    expect(wrapper.find(SpeciesCardType).length).toEqual(1)
    expect(wrapper.find(`img`).length).toEqual(0)
  })

  test(`uses a LatestExperimentCardContainer for latest experiment entries`, () => {
    console.log(randomExperimentTabName)
  	const wrapper = shallow(<PanelContent host={host} CardType={LatestExperimentCardType} resource={experimentResources} panelName={randomExperimentTabName}/>)
    expect(wrapper.find(LatestExperimentCardType).length).toEqual(1)
    expect(wrapper.find(SpeciesCardType).length).toEqual(0)
    expect(wrapper.find(`img`).length).toEqual(0)
  })

  test(`uses a static card for featured experiment`, () => {
  	const wrapper = shallow(<PanelContent host={host} CardType={LatestExperimentCardType} resource={experimentResources} panelName={`Featured`}/>)
    expect(wrapper.find(LatestExperimentCardType).length).toEqual(0)
    expect(wrapper.find(SpeciesCardType).length).toEqual(0)
    expect(wrapper.find(`img`).length).toEqual(1)
  })

  test(`matches snapshot`, () => {
    const tree = renderer.create(<PanelContent host={host} CardType={LatestExperimentCardType} resource={experimentResources} panelName={randomExperimentTabName}/>).toJSON()
    expect(tree).toMatchSnapshot()
  })

})
