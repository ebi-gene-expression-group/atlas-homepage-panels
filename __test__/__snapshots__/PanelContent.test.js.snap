// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelContent matches snapshot 1`] = `"I am a experiment component"`;
